state-chain/pallets/cf-swapping/src/tests/dca.rs

```diff
@@ -2,6 +2,100 @@ use super::*;
 
 const CHUNK_INTERVAL: u32 = 3;
 
+#[track_caller]
+fn setup_dca_swap(
+	number_of_chunks: u32,
+	chunk_interval: u32,
+	refund_params: Option<TestRefundParams>,
+) {
```

Comment:
It seems like a function like this could have been factored out more generally. 

There are checks here that are repeated eg. for FOK swaps (the sanity check, asserting on SwapRequested and SwapScheduled. It's not clear why we have a special-case function only for dca. Could it not just be `setup_swaps(&[TestSwapParams])`?

Reply:
It got messy when accounting for the differences between ccm and not ccm, I would have to add logic to handle the gas and do extra checks. But I do not like putting logic in test code. It's better to have explicit hard coded asserts without any logic.

Reply:
The dca test are quite different and only ever do one swap. The FoK tests use many swaps at a time and the setup code for them is already factored out and simple. Combining the setups makes both more complicated for no clear benefit.

Reply:
I disagree - the clear benefit is that the checks are consistently applied across all tests. I can see there some differences between the two and I don't know if this is deliberate and significant, or just because one copy of the code has been updated, but not the other.